workflow/views.py

```diff
@@ -24,28 +28,62 @@ def get_queryset(self):
 
 class Widget(viewsets.ModelViewSet):
     queryset = models.Widget.objects.all()
+
     serializer_class = serializers.Widget
 
     def get_queryset(self):
         return self.queryset
 
 
-class WidgetParameterMapping(viewsets.ModelViewSet):
-    queryset = models.WidgetParameterMapping.objects.all()
-    serializer_class = serializers.WidgetParameterMapping
+class ParameterAlias(viewsets.ModelViewSet):
+
+    queryset = models.ParameterAlias.objects.all()
+    serializer_class = serializers.ParameterAlias
 
     def get_queryset(self):
         return self.queryset
 
 
+class ParameterStub(viewsets.ModelViewSet):
+
+    queryset = models.ParameterStub.objects.all()
+    serializer_class = serializers.ParameterStub
+
+
 class Parameter(viewsets.ModelViewSet):
+
     queryset = models.Parameter.objects.all()
     serializer_class = serializers.Parameter
 
+    def retrieve(self, request, pk=None, case_pk=None):
```

Comment:
This lets us send queryparams that identify a unique record if we don't know the id of the record.